### PR TITLE
[SILInstruction] Introduce isDebugInstruction().

### PR DESCRIPTION
**Pull request by dcci**
This is a property of an instruction and should be a member
function of `SILInstruction` and not a free function in
`DebugUtils`. Discussed with Adrian.

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

File: include/swift/SIL/SILInstruction.h [hunk 1]
```diff
@@ -597,6 +597,14 @@ class SILInstruction
   /// you perform such optimizations like e.g. jump-threading.
   bool isTriviallyDuplicatable() const;
 
+  /// Returns true if the instruction \p Inst is an instruction which is
```
Comment (adrian-prantl) [inline include/swift/SIL/SILInstruction.h]: Copy and paste error: -> `... if this instruction ...`

Comment (adrian-prantl) [reply include/swift/SIL/SILInstruction.h]: lgtm otherwise.

**Comment by gottesmm** [comment]
@dcci why not just add a generic parent class and use isa.

**Comment by dcci** [comment]
@gottesmm I originally implemented that way, then figured out it's not really worth the additional boilerplate. If we want to go that route, I'd move `isMetaInstruction()` to that scheme as well, but I'm not entirely sure about the benefits. Happy to chat about it more.

**Comment by dcci** [comment]
@adrian-prantl fixed, thanks!

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
Ok. Sounds reasonable. I just wanted to make sure that you thought about that possibility.

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Comment by dcci** [comment]
@swift-ci smoke test and merge

**Pull request merged**